homeassistant/components/fordpass/__init__.py

```diff
@@ -0,0 +1,147 @@
+import logging
+
+import async_timeout
+from dotted.collection import DottedDict
+from fordpass import Vehicle
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, MANUFACTURER, VEHICLE, VIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["lock"]
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the FordPass component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up FordPass from a config entry."""
+    user = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    vin = entry.data[VIN]
+
+    coordinator = FordPassDataUpdateCoordinator(hass, user, password, vin)
+
+    await coordinator.async_refresh()  # Get initial data
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class FordPassDataUpdateCoordinator(DataUpdateCoordinator):
+    """DataUpdateCoordinator to handle fetching new data about the vehicle."""
+
+    def __init__(self, hass, user, password, vin):
+        """Initialize the coordinator and set up the Vehicle object."""
+        self._hass = hass
+        self.vin = vin
+        self.vehicle = Vehicle(user, password, vin)
+        self._available = True
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+
+    async def _async_update_data(self):
+        """Fetch data from FordPass."""
+        try:
+            async with async_timeout.timeout(30):
+                data = await self._hass.async_add_executor_job(
+                    self.vehicle.status  # Fetch new status
+                )
+
+                # If data has now been fetched but was previously unavailable, log and reset
+                if not self._available:
```

Comment:
Reading the code of DataUpdateCoordinator, it appears to have similar logic in async_refresh that logs a "Fetching %s data recovered" method, so i think this tracking is not needed.

Is it possible to use DataUpdateCoordinator without inheritence and instead use composition with  an async_update_data method like in this example? 
https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

or is there extra functionality here that can't use that?